lib/utils.ts

```diff
@@ -142,50 +142,48 @@ export function extendArray<T>(inArray: T[], otherArray: T[]) {
   }
 }
 
-/** 
- * Removes all items appearing in toSubtract from inArray, in-place 
+/**
+ * Removes all items appearing in toSubtract from inArray, in-place
  * @param toSubtract - The list of items to remove from inArray
  */
 export function subtractFromArray<T>(inArray: T[], toSubtract: T[]) {
   for (const value of toSubtract) {
-    inArray.splice(inArray.indexOf(value), 1);
+    removeFromArray(inArray, value);
```

Comment:
If `value` is not in `inArray` this would execure `inArray.splice(-1,1)`, removing the _last_ element in the array.

Reply:
😵Good point! I didn't catch this.